src/parser/Parser.ts

```diff
@@ -203,11 +203,8 @@ export class Parser {
             rightParen = advance();
 
             let maybeAs = peek();
-            if (check(Lexeme.Identifier) && maybeAs.text && maybeAs.text.toLowerCase() === "as") {
```

Comment:
No need to check if `maybeAs.text` is truthy before calling functions on it, since it's always defined and non-null! There's probably a lot of this around that'll get cleaned up eventually.